resolwe_bio/processes/import_data/expressions.py

```diff
@@ -0,0 +1,417 @@
+                ("DICTYBASE", "DICTYBASE"),
+                ("ENSEMBL", "ENSEMBL"),
+                ("NCBI", "NCBI"),
+                ("UCSC", "UCSC"),
+            ],
+        )
+        species = StringField(
+            label="Species",
+            description="Species latin name.",
+            allow_custom_choice=True,
+            choices=[
+                ("Homo sapiens", "Homo sapiens"),
+                ("Mus musculus", "Mus musculus"),
+                ("Rattus norvegicus", "Rattus norvegicus"),
+                ("Dictyostelium discoideum", "Dictyostelium discoideum"),
+                ("Odocoileus virginianus texanus", "Odocoileus virginianus texanus"),
+                ("Solanum tuberosum", "Solanum tuberosum"),
+            ],
+        )
+        build = StringField(
+            label="Build", description="Genome build or annotation version."
+        )
+        feature_type = StringField(
+            label="Feature type",
+            allow_custom_choice=True,
+            default="gene",
+            choices=[
+                ("gene", "gene"),
+                ("transcript", "transcript"),
+                ("exon", "exon"),
+            ],
+        )
+
+    class Output:
+        """Output fields to process UploadExpression."""
+
+        exp = FileField(label="Normalized expression")
+        rc = FileField(
+            label="Read counts",
+            required=False,
+            description="Reads mapped to genomic features.",
+        )
+        exp_json = JsonField(label="Expression (json)")
+        exp_type = StringField(label="Expression type")
+        exp_set = FileField(label="Expressions")
+        exp_set_json = JsonField(label="Expressions (json)")
+        source = StringField(label="Gene ID source")
+        species = StringField(label="Species")
+        build = StringField(label="Build")
+        feature_type = StringField(label="Feature type")
+
+    def run(self, inputs, outputs):
+        """Run analysis."""
+
+        if not inputs.exp and not inputs.rc:
+            self.error("Please provide raw or/and normalized expression files.")
+
+        elif inputs.exp and not inputs.exp_type:
+            self.error(
+                "Please provide normalization type together with normalized expressions."
+            )
+
+        elif not inputs.exp and inputs.exp_type and inputs.rc:
+            self.error("Please provide raw or/and normalized expression files.")
+
+        elif inputs.rc and not inputs.exp and not inputs.exp_type:
+            rc = inputs.rc.import_file(imported_format="compressed")
+            exp = inputs.rc.import_file(imported_format="compressed")
+            stem = Path(rc).stem
+            supported_extensions = (".txt", ".tab")
+            exp_type = "RAW_COUNT"
+
+        elif inputs.exp and inputs.exp_type and not inputs.rc:
+            exp = inputs.exp.import_file(imported_format="compressed")
+            stem = Path(exp).stem
+            supported_extensions = ".tab"
```

Comment:
Shouldn't here also be `(".txt", ".tab")`?

Reply:
In the old process only `".tab"` was the supported extension for exp, however if the supported extensions for exp and for rc are indeed not the same then I must add additional check in case where both, rc and exp are used as inputs.